### PR TITLE
Improve narrowing of CKEditor CDN configuration type

### PR DESCRIPTION
**Pull request by Mati365**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Feature: Improve result type of `useCKEditorCloud` to make `CKEditorPremiumFeatures` and `CKBox` non-nullable when proper configuration is passed.   
Fix: Rename the `languages` configuration property to `translations` in `useCKEditorCloud`.

MINOR BREAKING CHANGE: The `languages` configuration property has been renamed to `translations` in `useCKEditorCloud`.

---

### Additional information

Must be released before testing:  
https://github.com/ckeditor/ckeditor5-integrations-common/pull/25

PRs family:
https://github.com/ckeditor/ckeditor5-vue/pull/311

```tsx
const { CKEditorPremiumFeatures } = await loadCKEditorCloud( {
  version: '43.0.0',
  premium: true,
  ckbox: {
    version: '2.5.1'
  }
} );
```

**Before:**

`CKEditorPremiumFeatures` is nullable type.

**After**:

`CKEditorPremiumFeatures` is not nullable type because `premium: true` is passed.

### CODE REVIEW

File: tests/useMultiRootEditor.test.tsx [hunk 1]
```diff
@@ -9,8 +9,8 @@ import { describe, beforeEach, afterEach, it, expect, vi } from 'vitest';
 import React, { useEffect } from 'react';
 import { CKEditorError } from 'ckeditor5';
 import { render, waitFor, renderHook, act } from '@testing-library/react';
-import useMultiRootEditor, { EditorEditable, EditorToolbarWrapper } from '../src/useMultiRootEditor.tsx';
-import { ContextWatchdogContext } from '../src/context/ckeditorcontext.tsx';
+import useMultiRootEditor, { EditorEditable, EditorToolbarWrapper } from '../src/useMultiRootEditor';
+import { ContextWatchdogContext } from '../src/context/ckeditorcontext';
 import { timeout } from './_utils/timeout.js';
 import { createDefer } from './_utils/defer.js';
 import { createTestMultiRootWatchdog, TestMultiRootEditor } from './_utils/multirooteditor.js';
```
Comment (gorzelinski) [inline tests/useMultiRootEditor.test.tsx:16]: Those imports have extensions, and the other don't. I wonder if it's some convention or if we could unify it.
```suggestion
import { timeout } from './_utils/timeout';
import { createDefer } from './_utils/defer';
import { createTestMultiRootWatchdog, TestMultiRootEditor } from './_utils/multirooteditor';
```

File: tests/issues/39-frozen browser.test.tsx
Comment (gorzelinski) [inline tests/issues/39-frozen browser.test.tsx:1]: I would rename this file to `39-frozen-browser.test.tsx`

File: tests/cloud/useCKEditorCloud.test.tsx [hunk 2]
```diff
@@ -69,4 +69,81 @@ describe( 'useCKEditorCloud', () => {
...
+			await waitFor( () => {
+				expect( result.current.status ).toBe( 'success' );
+			} );
+
+			if ( result.current.status === 'success' ) {
+				expectTypeOf( result.current.CKBox ).not.toBeNullable();
+			}
+		} );
+
+		it( 'should return nullable ckbox entry type if ckbox not configured', async () => {
```
Comment (gorzelinski) [inline tests/cloud/useCKEditorCloud.test.tsx:135]:
```suggestion
		it( 'should return a nullable ckbox entry type if ckbox is not configured', async () => {
```

**Review by gorzelinski — COMMENTED** [review]
I added some nitpicks.

**Comment by Mati365** [comment]
@gorzelinski done

File: src/cloud/withCKEditorCloud.tsx [hunk 3]
```diff
@@ -24,7 +23,7 @@ import useCKEditorCloud from './useCKEditorCloud';
  * const withCKCloud = withCKEditorCloud( {
  * 	cloud: {
  * 		version: '42.0.0',
- * 		languages: [ 'en', 'de' ],
+ * 		translations: [ 'en', 'de' ],
```
Comment (filipsobol) [inline src/cloud/withCKEditorCloud.tsx:26]:
```suggestion
 * 		translations: [ 'es', 'de' ],
```

File: tests/cloud/useCKEditorCloud.test.tsx [hunk 4]
```diff
@@ -3,21 +3,21 @@
...
+import useCKEditorCloud from '../../src/cloud/useCKEditorCloud.js';
 
 describe( 'useCKEditorCloud', () => {
 	beforeEach( removeAllCkCdnResources );
 
 	it( 'should load CKEditor bundles from CDN', async () => {
 		const { result } = renderHook( () => useCKEditorCloud( {
 			version: '43.0.0',
-			languages: [ 'en', 'de' ]
+			translations: [ 'en', 'de' ]
```
Comment (filipsobol) [inline tests/cloud/useCKEditorCloud.test.tsx:20]:
```suggestion
			translations: [ 'es', 'de' ]
```

I'm not sure about this

File: tests/index.test.tsx [hunk 5]
```diff
@@ -7,10 +7,10 @@ import { describe, afterEach, it, expect, vi } from 'vitest';
...
 import { ContextWatchdog } from 'ckeditor5';
 import { render, type RenderResult } from '@testing-library/react';
-import ContextMock from './_utils/context.js';
-import Editor from './_utils/editor.js';
-import { PromiseManager } from './_utils/promisemanager.js';
-import { CKEditor, CKEditorContext } from '../src/index.js';
+import ContextMock from './_utils/context';
+import Editor from './_utils/editor';
+import { PromiseManager } from './_utils/promisemanager';
+import { CKEditor, CKEditorContext } from '../src/index';
```
Comment (filipsobol) [inline tests/index.test.tsx:13]: Missing file extensions

File: tests/integrations/ckeditor-classiceditor.test.tsx [hunk 6]
```diff
@@ -10,11 +10,11 @@ import React, { createRef } from 'react';
 import ReactDOM from 'react-dom';
 import { render, type RenderResult } from '@testing-library/react';
 
-import CKEditor from '../../src/ckeditor.tsx';
+import CKEditor from '../../src/ckeditor';
 
-import { timeout } from '../_utils/timeout.js';
+import { timeout } from '../_utils/timeout';
```
Comment (filipsobol) [inline tests/integrations/ckeditor-classiceditor.test.tsx]: Missing file extensions

**Review by filipsobol — APPROVED** [review]

**Pull request merged**